clang/test/CodeGenSYCL/kernel-caller-entry-point.cpp

```diff
@@ -100,11 +100,8 @@ int main() {
 
 // Verify that SYCL kernel caller functions are emitted for each device target.
 //
-// FIXME: The following set of matches are used to skip over the declaration of
-// main(). main() shouldn't be emitted in device code, but that pruning isn't
-// performed yet.
-// CHECK-DEVICE:      Function Attrs: convergent mustprogress noinline norecurse nounwind optnone
-// CHECK-DEVICE-NEXT: define {{[a-z_ ]*}}noundef i32 @main() #0
+// main() shouldn't be emitted in device code.
+// CHECK-NOT: define {{[a-z_ ]*}}noundef i32 @main() #0
```

Comment:
Can this be simplified to be less sensitive to the presence of keywords? Can we just look for `@main()`?
```suggestion
// CHECK-NOT: @main()
```